src/commands/mod.rs

```diff
@@ -580,13 +580,23 @@ impl DaemonControl {
         }
     }
 
-    pub fn list_spend(&self) -> ListSpendResult {
+    pub fn list_spend(&self, txids: Option<Vec<bitcoin::Txid>>) -> ListSpendResult {
         let mut db_conn = self.db.connection();
-        let spend_txs = db_conn
-            .list_spend()
+        let spend_psbts = db_conn.list_spend();
+
+        let txids_set = txids
+            .as_ref()
+            .map(|list| list.iter().cloned().collect::<HashSet<_>>())
+            .unwrap_or_default();
+
+        let spend_txs: Vec<ListSpendEntry> = spend_psbts
```

Comment:
nit:
```suggestion
        let spend_txs = spend_psbts
```

(deduced from the returned type)